### PR TITLE
Fix convert camel to dashed

### PR DESCRIPTION
**Pull request by KateWilkinson**
Conversion of camel cased strings into dash-separated strings was not working in every case - in particular when strings include two letter words. The following commit shows the issue and the workaround currently needed: https://github.com/moneyadviceservice/common_initial_assessment/commit/8bfa913b7d2257bb336e3f8a8004828f265a5acd

* More test permutations added to ensure strings with words of different lengths are being tested for.
* Regex in convertCamelCaseToDashed reworked to ensure all permutations are now being converted accurately.

### CODE REVIEW

File: spec/js/tests/utilities_spec.js [hunk 1]
```diff
@@ -40,10 +40,92 @@ describe('utilities', function() {
...
+      ['ABCAbcAbc', 'abc-abc-abc'],
+      ['AbcABCAbc', 'abc-abc-abc'],
+      ['AbcAbcABC', 'abc-abc-abc'],
+
+      // Examples (from the original test)
+      ['FooBar', 'foo-bar'],
+      ['FOOBar', 'foo-bar'],
+      ['FooBarBaz', 'foo-bar-baz'],
+
+      // Actual component names (to catch regressions)
```
Comment (KateWilkinson) [inline spec/js/tests/utilities_spec.js]: Actual component names were added to the test here for extra security while working out the bug - how do people feel about these? Are they all needed?

File: spec/js/tests/utilities_spec.js [hunk 2]
```diff
@@ -40,10 +40,92 @@ describe('utilities', function() {
...
+      ['ScrollTo', 'scroll-to'],
+      ['ShowMore', 'show-more'],
+      ['StickyColumn', 'sticky-column'],
+      ['TabSelector', 'tab-selector'],
+      ['TabularTooltip', 'tabular-tooltip'],
+      ['ToggleField', 'toggle-field'],
+      ['ToggleHint', 'toggle-hint'],
+      ['Tooltip', 'tooltip'],
+      ['Validation', 'validation'],
+      ['ViewAll', 'view-all'],
```
Comment (KateWilkinson) [inline spec/js/tests/utilities_spec.js]: Actual component names were added to the test here for extra security while working out the bug - how do people feel about these? Are they all needed?

**Comment by munckymagik** [comment]
FYI the CodeClimate fail was this:

screen shot 2016-01-27 at 10 12 09

**Comment by mconnell** [comment]
The other examples for your test appear to cover all the permutations, but it doesn't hurt to have the real names, especially as it's a change which would impact on all of the components. :+1:

**Comment by urfolomeus** [comment]
I concur with Mark :+1:

**Comment by davidtrussler** [comment]
:+1: LGTM

**Pull request merged**